packages/agreement-process/src/model/domain/validators.ts

```diff
@@ -69,6 +69,18 @@ export const assertRequesterIsConsumer = (
   }
 };
 
+export const assertRequesterIsConsumerOrProducer = (
+  agreement: Agreement,
+  authData: AuthData
+): void => {
+  if (
+    authData.organizationId !== agreement.consumerId &&
+    authData.organizationId !== agreement.producerId
+  ) {
+    throw operationNotAllowed(authData.organizationId);
+  }
+};
```

Comment:
The method name looks like an OR condition but is implemented with AND condition.
Any way in the `main` branch already exists the method `assertRequesterIsConsumerOrProducer` it should be useful for suspend workflow.
https://github.com/pagopa/interop-be-monorepo/blob/22741274ef00c7f8ac7fad9184b44477829c5c0b/packages/agreement-process/src/model/domain/validators.ts#L82

```suggestion
```

Reply:
Rebased and reused the one on `main`!

Reply:
👍🏻  ❤️